src/pupremote.ts

```diff
@@ -37,12 +37,12 @@ export class PUPRemote extends LPF2Hub {
     protected _voltageMaxRaw = 3200;
 
 
-    constructor (device: IBLEDevice, autoSubscribe: boolean = true) {
+    constructor (device: IBLEAbstraction, autoSubscribe: boolean = true) {
         super(device, autoSubscribe);
-        this.type = Consts.HubType.POWERED_UP_REMOTE;
-        this._ports = {
-            "LEFT": new Port("LEFT", 0),
-            "RIGHT": new Port("RIGHT", 1)
+        this._type = Consts.HubType.POWERED_UP_REMOTE;
+        this._portNames = {
+            "LEFT": 0,
+            "RIGHT": 1
```

Comment:
This looks repetitive over hubs, does it make sense to use the same approach you have on devices ?

```js
super(device, autosubscribe, Consts.HubType.POWERED_UP_REMOTE, {
    "LEFT": 0,
    "RIGHT": 1
});
```

Reply:
Yeah, perhaps so. To be honest I’ve barely touched the hub setup, I’ve been focused on the device abstraction. But they will get a revamp too.

Reply:
This is also now done - https://github.com/nathankellenicki/node-poweredup/pull/61#discussion_r359069774